artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2550,6 +2552,11 @@ public boolean isExpirationRedundant() {
          return true;
       }
 
+      if (isInternalQueue() && name.toString().startsWith(MIRROR_ADDRESS)) {
+         logger.trace("Queue {} is not supposed to expiry as it is a Mirror Store and Forward Queue", this.name);
```

Comment:
The "not supposed to expiry" does not really make sense, which is partly why I changed it to "expire messages" in my previous updated suggestion.